book/ecosystem.ipynb

```diff
@@ -0,0 +1,260 @@
+    "# Display the output\n",
+    "fig.show()\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "b73a6666-6c2c-4f49-a92b-995ade576ccc",
+   "metadata": {},
+   "source": [
+    "## Interactive data visualization with Xarray, Panel, and PyGMT"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "1bf15df6-6a4f-4221-af66-0db1c5b1e328",
+   "metadata": {},
+   "source": [
+    "In this section, we'll create some interactive visualizations of oceanographic data!\n",
+    "\n",
+    "We'll use [Panel](https://panel.holoviz.org/index.html), which is a Python library\n",
+    "for connecting interactive widgets with plots! We'll use Panel with\n",
+    "[PyGMT](https://www.pygmt.org) and [xarray](https://www.xarray.dev) to visualize\n",
+    "the objectively interpolated mean field for in-situ temperature from the World Ocean Atlas.\n",
+    "\n",
+    "**References**:\n",
+    "\n",
+    "- Temperature visualization based on https://rabernat.github.io/intro_to_physical_oceanography/02-c_ocean_temperature_salinity_stratification.html\n",
+    "- Interactive setup based on https://github.com/weiji14/30DayMapChallenge2021/blob/main/day25_interactive.py\n",
+    "- Data from the NOAA World Ocean Atlas, stored on the IRI Data Library at http://iridl.ldeo.columbia.edu/SOURCES/.NOAA/.NODC/.WOA09/."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "dbcff01f-fac0-4c1d-bb66-89dcb2e7711b",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import panel as pn\n",
+    "import xarray as xr\n",
+    "import pygmt\n",
+    "pn.extension()\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "7b2f9cb8-add6-45c2-9f54-8e33ed9e6f02",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Download the dataset from the IRI Data Library\n",
+    "url = 'https://iridl.ldeo.columbia.edu/SOURCES/.NOAA/.NODC/.WOA09/.Grid-1x1/.Annual/.temperature/.t_an/data.nc'\n",
+    "netcdf_file = pygmt.which(fname=url, download=True)\n",
+    "woa_temp = xr.open_dataset(netcdf_file)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "b11ea510-7540-475e-9217-be4204cb1ea3",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Make a panel widget for controlling the depth plotted\n",
+    "depth_slider = pn.widgets.DiscreteSlider(name='Discrete Slider', options=woa_temp.coords['depth'].values.astype(int).tolist(), value=0)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "3496d4db-92d9-45ee-9d48-6b4f00a19361",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Make a function for plotting the depth slice with PyGMT\n",
+    "\n",
+    "@pn.depends(depth=depth_slider)\n",
+    "def view(depth: int):\n",
+    "    fig = pygmt.Figure()\n",
+    "    pygmt.makecpt(cmap=\"vik\", series=[-2,30])\n",
+    "    fig.grdimage(grid=woa_temp.t_an.sel(depth=depth).isel(time=0), cmap=True, projection=\"R15c\", frame=True)\n",
```

Comment:
Minor suggestion, but was thinking whether the `.isel(time=0)` could be done in one of the cells above? It might make the rendering slightly faster to not have the do this slicing everytime the slider is dragged.